R/readepi_helpers.R

```diff
@@ -130,3 +128,40 @@ fingertips_get_args <- function(args_list =
     parent_area_type_id = parent_area_type_id
   )
 }
+
+#' Check if the base_url argument is specified as a correct URL
+#'
+#' @param base_url the URL to the HIS of interest
+#'
+#' @keywords internal
+#'
+url_check <- function(base_url) {
```

Comment:
Wouldn't it be easier to test all of this with a single regex?

```r
  stopifnot("Incorrect domain name in provided URL!" =
              grepl("^(https?://)?[^.]\\.[^.]", base_url))
```

Note that this is not a perfect regex, a lot of invalid cases will still pass but it should be equivalent to what you have at the moment.

Reply:
What about this comment @Karim-Mane?

Reply:
The correct regex is this. Could you give it another go please?

```r
grepl("^(https?://)?[^.]+\\.[^.]+", "https://karim.sn") 
```

Reply:
It works as expected - thanks